python/ray/data/_internal/dataset_logger.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import os
+from typing import Optional
+
+import ray
+from ray._private.ray_constants import LOGGER_FORMAT
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class DatasetLogger:
+    """Logger for Ray Datasets which, in addition to logging to stdout,
+    also writes to a separate log file at `DatasetLogger.DEFAULT_DATASET_LOG_PATH`.
+    """
+
+    DEFAULT_DATASET_LOG_PATH = "logs/datasets.log"
+
+    def __init__(self, log_name, dataset_log_path: Optional[str] = None):
+        """Initialize DatasetLogger for a given `log_name`.
+
+        Args:
+            log_name: Name of logger (usually passed into `logging.getLogger(...)`)
+            dataset_log_path: If specified, write logs to this path
+            instead of `DatasetLogger.DEFAULT_DATASET_LOG_PATH`
+        """
+        self.logger = logging.getLogger(log_name)
+        self.logger.setLevel(logging.INFO)
+
+        # If ray.init() was previously called, add log handler which writes to
+        # a separate Datasets log file at `DatasetLogger.DEFAULT_DATASET_LOG_PATH`
+        if ray.is_initialized():
+            # With current implementation, we can only get session_dir
+            # after ray.init() is called. Less hacky way could potentially fix this
+            session_dir = ray._private.worker._global_node.get_session_dir_path()
```

Comment:
[Some of the tests](https://buildkite.com/ray-project/oss-ci-build-pr/builds/6138) are not happy with this approach. Is there another way to work around this?

Reply:
Can we just check if ray._private.worker._global_node is not None?

Similar code path - https://github.com/ray-project/ray/blob/cfee6d09a0baf6ed28d0642481330fae683e732e/python/ray/cluster_utils.py#L245-L246